src/secp256k1.c

```diff
@@ -605,50 +612,58 @@ int secp256k1_ec_pubkey_negate(const secp256k1_context* ctx, secp256k1_pubkey *p
     return ret;
 }
 
-int secp256k1_ec_seckey_tweak_add(const secp256k1_context* ctx, unsigned char *seckey, const unsigned char *tweak) {
+
+static int secp256k1_ec_seckey_tweak_add_helper(secp256k1_scalar *sec, const unsigned char *tweak) {
     secp256k1_scalar term;
+    int overflow = 0;
+    int ret = 0;
+
+    secp256k1_scalar_set_b32(&term, tweak, &overflow);
+    ret = (!overflow) & secp256k1_eckey_privkey_tweak_add(sec, &term);
+    secp256k1_scalar_clear(&term);
+    return ret;
+}
+
+int secp256k1_ec_seckey_tweak_add(const secp256k1_context* ctx, unsigned char *seckey, const unsigned char *tweak) {
     secp256k1_scalar sec;
     int ret = 0;
-    int overflow = 0;
     VERIFY_CHECK(ctx != NULL);
     ARG_CHECK(seckey != NULL);
     ARG_CHECK(tweak != NULL);
 
-    secp256k1_scalar_set_b32(&term, tweak, &overflow);
     ret = secp256k1_scalar_set_b32_seckey(&sec, seckey);
-
-    ret &= (!overflow) & secp256k1_eckey_privkey_tweak_add(&sec, &term);
+    ret &= secp256k1_ec_seckey_tweak_add_helper(&sec, tweak);
     secp256k1_scalar_cmov(&sec, &secp256k1_scalar_zero, !ret);
```

Comment:
Should the `secp256k1_scalar_cmov` be part of `secp256k1_ec_seckey_tweak_add_helper`?

Reply:
Currently that `cmov` takes into account whether `scalar_set_b32_seckey` failed because it's part of `ret`. If `cmov` was in the helper then the function can fail due to the seckey overflowing the scalar, but the seckey wouldn't be zeroed after the function call.

Reply:
Yes of course.